sympy/solvers/bivariate.py

```diff
@@ -199,17 +244,61 @@ def _solve_lambert(f, symbol, gens):
     nrhs, lhs = f.as_independent(symbol, as_Add=True)
     rhs = -nrhs
 
+    even_degrees = [i for i in list(set(flatten(power_list(lhs, symbol)))) \
+                    if i%2 == 0 and i != 0]
+    t = Dummy('t', **symbol.assumptions0)
+
     lamcheck = [tmp for tmp in gens
                 if (tmp.func in [exp, log] or
                 (tmp.is_Pow and symbol in tmp.exp.free_symbols))]
     if not lamcheck:
         raise NotImplementedError()
 
+    def _solve_even_degree_branches(diff, symbol, t, gens):
+        """
+        Helper function to consider even degree branches of lambert
+        equation if exists.
+
+        For example:
+        Equation (a/x + exp(x/2)).diff(x) = 0 is represented
+        in logform as both of the below equations.
+        1. 2*log(x) + x/2 - log(a) = 0
+        2. 2*log(-x) + x/2 - log(a) = 0
+
+        Input: diff, symbol, t, gens
+        diff - Includes a dummy variable t which can be replaced
+        with +symbol and -symbol.
+
+        Output: Returns combined solution after considering all
+        even degree branches of original equation.
+        """
+        llhs1, llhs2 = map(lambda i: diff.xreplace({t: i}), (symbol, -symbol))
+        if llhs1 == llhs2:
```

Comment:
Which equation(s) falls here? Add it in the tests (if not added yet).

Reply:
```python
>>> solve((x**2 - 2*x + 1).subs(x, log(x) + 3*x))
[LambertW(3*E)/3]
# llhs1, llhs2 comes out to be same 
# -log(log(-(3*x - 1)**2 - (3*x + log(x))**2 + 2*log(x)) + zoo) + zoo == -log(log(-(3*x - 1)**2 - (3*x + # log(x))**2 + 2*log(x)) + zoo) + zoo
```
Already added.